workspaces/arborist/lib/arborist/load-virtual.js

```diff
@@ -200,6 +200,13 @@ module.exports = cls => class VirtualLoader extends cls {
       const targetPath = resolve(this.path, meta.resolved)
       const targetLoc = relpath(this.path, targetPath)
       const target = nodes.get(targetLoc)
+
+      if (!target) {
+        const err = new Error(`Missing target in lock file: "${targetLoc}" is referenced by "${location}" but does not exist.\nThis is likely due to a damaged lock file. Remove the entry for "${location}" from your lock file and run npm install again.`)
```

Comment:
Typically we don't suggest folks editing their lockfile manually.  Is there a better programmatic way to fix this?

Reply:
@wraithgar fair point! I updated it. See what you think 😄